packages/stokhos/src/sacado/kokkos/vector/belos/Belos_TpetraAdapter_MP_Vector.hpp

```diff
@@ -355,8 +355,8 @@ namespace Belos {
       typedef Tpetra::MultiVector<dot_type,LO,GO,Node> FMV;
       typedef typename FMV::dual_view_type::t_dev flat_view_type;
       typedef typename flat_view_type::execution_space execution_space;
-      flat_view_type flat_A_view = Atmp->template getLocalView<execution_space>();
-      flat_view_type flat_C_view = Ctmp->template getLocalView<execution_space>();
+      flat_view_type flat_A_view = const_cast<MV*>(Atmp.get())->getLocalViewDevice(Tpetra::Access::ReadWrite);
+      flat_view_type flat_C_view = const_cast<MV*>(Ctmp.get())->getLocalViewDevice(Tpetra::Access::ReadWrite);
```

Comment:
A bit weird that the `Atmp` and `Ctmp` are declared `RCP<const MV>` since we want to write into their local device view...

Maybe this fix could just be `Atmp->getLocalViewDevice(Tpetra::Access::ReadWrite)` and change the type of `Atmp` to `RCP<MV>` ?

Reply:
Actually, I believe C is overwritten but A is not.  So `Atmp` should be `RCP<const MV>` and `Ctmp` should be `RCP<M>`, in which case I think it should be
```
typedef typename MV::dual_view_type::t_dev const_flat_view_type;
const_flat_view_type flat_A_view = Atmp->getLocalViewDevice(Tpetra::Access::ReadOnly);
flat_view_type flat_C_view = Ctmp->getLocalViewDevice(Tpetra::Access::ReadWrite);
```

Reply:
We introduced PR
* https://github.com/trilinos/Trilinos/pull/10710

with these changes :smile: So I'll wait it is accepted to rebase on it.